tests/contracts/priceSources/MockChaiPriceSource.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.6.8;
+
+contract MockChaiPriceSource {
+    uint256 public chi;
+    uint256 public rho;
+    uint256 public drip;
```

Comment:
I'd be interested to see if this works, because because `drip()` is a non-view function that updates state.

I think the best thing to do is to get rid of this and `setDrip()`, and define a `function drip() external` that does nothing. (or if you're feeling ambitious and want to actually have it update `chi` and `rho`, you could do that also! Not sure if that's necessary for unit tests though?)

Here's the actual `drip()` function: https://github.com/makerdao/dss/blob/b280b98e0c661d5d8effe2e9626e51978dab33ce/src/pot.sol#L140

And this is the `Pot` contract documentation in case that's helpful: https://docs.makerdao.com/smart-contract-modules/rates-module/pot-detailed-documentation

Reply:
If you can give this another review before you start your deep dive in the upgradability stuff that would be great. Once you you give the green light for the mock contracts I'll start with the tests.